src/nodedef.h

```diff
@@ -505,8 +525,13 @@ struct ContentFeatures
 	}
 
 #if CHECK_CLIENT_BUILD()
+	void preUpdateTextures(ITextureSource *tsrc,
+		std::unordered_set<std::string> &pool, const TextureSettings &tsettings);
 	void updateTextures(ITextureSource *tsrc, IShaderSource *shdsrc,
-		scene::IMeshManipulator *meshmanip, Client *client, const TextureSettings &tsettings);
+		Client *client, PreLoadedTextures *texture_pool,
+		const TextureSettings &tsettings);
+	void updateMesh(Client *client, const TextureSettings &tsettings);
+	void collectMaterials(std::vector<u32> &leaves_materials);
```

Comment:
collectLeaveMaterials

Reply:
It too much discussion about this already :)
I thought this is specific to collect leave materials and so I thought it might be good to name the function as such. That's all. It's not important.